### PR TITLE
Improve performance of emissary.util.Hexl.toUnformattedHexString()

### PR DESCRIPTION
**Pull request by jdcove2**

### CODE REVIEW

File: src/main/java/emissary/util/Hexl.java [hunk 1]
```diff
@@ -14,24 +14,37 @@
...
+            }
+        }
+    }
+
     /**
      * Unformatted hex string
      */
     public static String toUnformattedHexString(byte[] data) {
-        StringBuilder output = new StringBuilder(2048);
+        final char[] characters = new char[data.length * 2];
```
Comment (fbruton) [inline src/main/java/emissary/util/Hexl.java:37]: Recommend comparing performance against ```org.apache.commons.codec.binary.Hex.encodeHexString```. If the performance differences are negligible, I recommend going with the Apache commons implementation.

**Review by fbruton — CHANGES_REQUESTED** [review]
Request change based on previous comment.

File: src/main/java/emissary/util/Hexl.java (hunk 1 above)
Comment (jdcove2) [reply src/main/java/emissary/util/Hexl.java]: The new code and the encodeHexString method are practically the same so their performance should be practically the same; therefore, I have replaced the new code with the encodeHexString method.

**Review by fbruton — APPROVED** [review]

**Comment by jdcove2** [comment]
The following screenshots show the number of times and memory allocated for processing a single 20Meg mysqldump file. The first screenshot shows the number & memory created by the original code. The second screenshot shows the number & memory created by the new code.

image

image

**Review by jpdahlke — COMMENTED** [review]
LGTM, but....

Did we want to take this one step further? If we are just going to call the apache commons library, can we just update all references to this method with a call the apache commons version? It appears to only be a handful. It's a breaking change but we haven't been very good about maintaining that in v6 so far....

Also, this puts the usage of the class and its methods on the brink. I would recommend deprecating this class in the pr at the least and then we can consider removal in a follow-up.

**Comment by jdcove2** [comment]
The code cleanup suggestion will be performed in a subsequent PR.

**Comment by jpdahlke** [comment]
https://github.com/NationalSecurityAgency/burrito-grande/issues/1463

**Review by jpdahlke — APPROVED** [review]

**Pull request merged**